docs/design/eth-filter.md

```diff
@@ -0,0 +1,133 @@
+# Filter API design
+
+## Purpose
+
+The JSON-RPC Relay currently doesn't support monitoring events using filter API as is common in many other relays and is expected by many web3 tools, which makes polling for event logs, new blocks and transactions hard for developers.
+
+## Goals
+
+1. Introduce new methods:
+   1. `eth_newFilter`
+   2. `eth_uninstallFilter`
+   3. `eth_getFilterChanges`
+   4. `eth_getFilterLogs`
+
+## Non-Goals
+
+1. Decrease the load on the subscription, as this is an alternative.
+2. Increase the supported functionality of the relay.
+
+## Architecture
+
+New filter API methods can be added and handled by extending `eth` class. Adding several new methods is needed and for saving the filter IDs it needs to utilize the already available cache.
+
+### Filter Types
+
+|      Filter Type       |                                                    Description                                                     |                           Support                           |
+| :--------------------: | :----------------------------------------------------------------------------------------------------------------: | :---------------------------------------------------------: |
+|          logs          |                               Returns all logs matching filter with given filter ID.                               |                       Yes, initially                        |
+|       newBlocks        |                            Returns all new blocks matching filter with given filter ID.                            |                 No, maybe in future release                 |
```

Comment:
Q: Do we know how often this is utilized in other RPCs?

Reply:
Didn't find any information regarding this.